reference/tests/AddOpTest.cpp

```diff
@@ -0,0 +1,290 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+   Copyright 2022 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "gtest/gtest.h"
+#include "reference/tests/RunTestUtils.h"
+
+namespace mlir {
+namespace stablehlo {
+
+TEST(AddOpInterpreterTest, F16) {
+  constexpr llvm::StringRef kModule = R"(
+  module {
+    func.func @main(%arg0: tensor<2x5xf16>, %arg1: tensor<2x5xf16>) -> tensor<2x5xf16> {
+      %result = stablehlo.add %arg0, %arg1 : tensor<2x5xf16>
+      func.return %result : tensor<2x5xf16>
+    }
+  })";
+  runTestCase(kModule, {/*operands*/
+                        {"0.0", "-0.0", "1.0", "0.125", "0.1", "3.141", "inf",
+                         "inf", "-inf", "inf"},
+                        {"0.0", "-0.0", "7.0", "0.75", "0.3", "3.141", "0",
+                         "inf", "-inf", "-inf"},
+                        /*expected result*/
+                        {"0.0", "-0.0", "8.0", "0.875", "0.4", "6.282", "inf",
+                         "inf", "-inf", "nan"}});
+}
+
+TEST(AddOpInterpreterTest, F32) {
```

Comment:
Reorder test cases so that F32 comes after BF16, to be consistent with the order everywhere else?